core/application/notificationWorkflow/NotificationService.ts

```diff
@@ -64,7 +64,37 @@ export class NotificationService {
       }
       await userRepository.update(updateData)
       return 'Device registration successful.'
-    } catch (error) {
+    } catch (error: unknown) {
+      const typedError = error as Error
+      if (
+        typedError.name === 'InvalidParameterException' &&
+        typedError.message.includes('already exists with the same Token')
```

Comment:
The message can be a little more specific like `Device already registered with this Token`